resources/git-rebase.md

```diff
@@ -0,0 +1,83 @@
+# Git rebase
+
+## What’s rebasing?
+
+Git rebase takes a branch you select and merges it on to the branch where you’re currently working. Then it takes all of your local commits and applies them on top.
+
+## You said it merges the branches. Why not use `git merge`?
+
+Git rebase solves the same issues as git merge, but rebasing results in a cleaner, easier to understand commit history (better for everyone trying to understand the history of your project!). Rebasing keeps your history clean by moving your commits to the end of your git history, while git merge makes a merge commit every time you pull upstream changes, but keeps your commits at the same at the same point in history when they were made (resulting in a strange spliting of your changes in your git history). 
+
+tl;dr: git rebase makes your commit history easier to read than git merge. To learn more about the difference between merging and rebasing (with handy diagrams!), check out this link: https://www.atlassian.com/git/tutorials/merging-vs-rebasing  
+
+## When should I rebase?
+
+One common reason for rebasing is to get changes from another branch and apply them to your own. Let’s say you and your team are busy working on your project. You’re coding on your local branch when someone else on your team finishes a feature and merges it to master. The new code your friend just pushed is a change you want on your local branch. You can use `git rebase <nameOfYourFriendsBranch>` to add these changes to your local branch and keep working.
+
+Another reason to rebase is to head off (or handle!) any potential merge conflicts. A merge conflict occurs when two branches have code edits on the same lines. Git doesn’t know which of the edits should be kept, so it throws a conflict that a human (you!) has to resolve. So if a teammate's branch is merged before yours and you and your teammate have both touched the same files, your branch could now have merge conflicts. The fix? Rebasing.
```

Comment:
Maybe clarify here at the end, that the fix is rebasing and rebasing often. (as you'll likely always have conflicts, it's just easier to deal with when you do a bit at a time

Reply:
Nevermind! Ignore me, just saw you said this in the next section :D